vitrina/datasets/views.py

```diff
@@ -3,180 +3,62 @@
 
 from django.conf import settings
 from django.contrib.auth.mixins import LoginRequiredMixin, PermissionRequiredMixin
-from django.db.models import Q
 from django.http import FileResponse, JsonResponse
-from django.http import HttpResponseBadRequest
 from django.shortcuts import get_object_or_404
+from django.views import View
+from django.views.generic import TemplateView
 from django.shortcuts import redirect
 from django.utils.translation import gettext_lazy as _
-from django.views import View
-from django.views.generic import ListView, TemplateView
 from django.views.generic.detail import DetailView
 from django.views.generic.edit import CreateView, UpdateView
+from haystack.generic_views import FacetedSearchView
 
 # TODO: I think, Django has this built-in.
 from slugify import slugify
 
 from vitrina.classifiers.models import Category
 from vitrina.classifiers.models import Frequency
-from vitrina.datasets.forms import DatasetFilterForm
 from vitrina.datasets.forms import NewDatasetForm
+from vitrina.datasets.forms import DatasetSearchForm
+from vitrina.helpers import get_selected_value
 from vitrina.datasets.models import Dataset, DatasetStructure
-from vitrina.datasets.services import can_create_dataset
-from vitrina.datasets.services import can_update_dataset
-from vitrina.datasets.services import filter_by_status
-from vitrina.datasets.services import get_category_counts
-from vitrina.datasets.services import get_related_categories
-from vitrina.datasets.services import get_related_tag_list
-from vitrina.datasets.services import get_tag_list
-from vitrina.helpers import get_selected_value, get_filter_url
+from vitrina.datasets.services import update_facet_data, can_update_dataset, can_create_dataset
 from vitrina.orgs.helpers import is_org_dataset_list
 from vitrina.orgs.models import Organization
 from vitrina.orgs.services import has_coordinator_permission
 from vitrina.resources.models import DatasetDistribution
 
 
-class DatasetListView(ListView):
-    model = Dataset
+class DatasetListView(FacetedSearchView):
     template_name = 'vitrina/datasets/list.html'
+    facet_fields = ['filter_status', 'organization', 'category', 'frequency', 'tags', 'formats']
+    form_class = DatasetSearchForm
     paginate_by = 20
 
-    def get(self, request, *args, **kwargs):
-        filter_form = DatasetFilterForm(self.request.GET)
-        if filter_form.is_valid():
-            return super().get(request, *args, **kwargs)
-        return HttpResponseBadRequest(_("Klaida"))
-
     def get_queryset(self):
-        datasets = Dataset.public.order_by('-published')
-        if is_org_dataset_list(self.request):
-            organization = get_object_or_404(
-                Organization,
-                pk=self.kwargs['pk'],
-            )
-            datasets = datasets.filter(organization=organization)
```

Comment:
By removing this, you broke `organization-datasets` view, could you add tests for this case and fix the issue?

Currently `organization-datasets` does not filter by organization, but shows all datasets, from all organizations.

Reply:
This is still not fixed.